modules/photo-app/reducers/galleries.js

```diff
@@ -11,33 +11,43 @@ const initialState = {
 
 const galleries = (state = initialState, action) => {
   switch (action.type) {
-    case 'LOAD_GALLERY_DATA_REQUEST':
+    case 'LOAD_GALLERY_DATA_REQUEST': {
+      const galleryId = action.payload.id;
+
+      return {                   // return brand new object
+        ...state,                // containing all properties of the `state` object
+        [galleryId]: {           // but also adding (or overwriting if already exists) property with name held by `galleryId` which points to new object
+           ...state[galleryId],  // that contains all properties of the object referenced by the same name in the `state` object
+           fetching: true,       // but also adding `fetching` property with value `true`
+        }
+      };
+    }
+
+    case 'LOAD_GALLERY_DATA_SUCCESS': {
+      const galleryId = action.payload.id;
+
       return {
         ...state,
-        [action.payload.id]: state[action.payload.id],
-        fetching: true,
-      };
-      // Object.assign({}, state, {
-      //   [action.payload.id]: Object.assign({}, state[action.payload.id], {
-      //     fetching: true,
-      //   }),
-      // });
-
-    case 'LOAD_GALLERY_DATA_SUCCESS':
-      return Object.assign({}, state, {
-        [action.payload.id]: Object.assign({}, state[action.payload.id], {
+        [galleryId]: {
+          ...state[galleryId],
           photos: normalizePhotos(action.payload.data.photo),
           fetching: false,
           lastFetched: Date.now(),
-        }),
-      });
+        }
+      };
+    }
+
+    case 'LOAD_GALLERY_DATA_FAILURE': {
```

Comment:
You might notice that I wrapped `case`s into curly braces: in this context they simply create new code blocks, so I use them to isolate `let`s and `const`s and reuse same name (`galleryId`) in several reducer cases (otherwise they'd conflict since they share the same block/function scope).